addons/survey/tests/test_survey.py

```diff
@@ -681,3 +695,105 @@ def assertAnswerStatus(self, expected_answer_status, questions_statistics):
+            {'value': 'A thing full of letters.', 'answer_score': 1.0},
+            {'value': 'A unit of language, [...], carrying a meaning.', 'answer_score': 4.0, 'is_correct': True},
+            {'value': 'A thing related to space', 'answer_score': -4.0},
+            {'value': 'Yes', 'answer_score': -0.5},
+            {'value': 'No', 'answer_score': 0.5, 'is_correct': True},
+            {'value': 'Yes', 'answer_score': 0.5, 'is_correct': True},
+            {'value': 'No', 'answer_score': 0.2},
+        ])
+        q1, q2, q3 = self.env['survey.question'].create([{
+            'survey_id': survey.id,
+            'title': 'What is a word?',
+            'sequence': 2,
+            'question_type': 'simple_choice',
+            'suggested_answer_ids': [Command.set((a_q1_partial | a_q1_correct | a_q1_incorrect).ids)],
+            'constr_mandatory': False,
+        }, {
+            'survey_id': survey.id,
+            'title': 'Are you sure?',
+            'sequence': 3,
+            'question_type': 'simple_choice',
+            'suggested_answer_ids': [Command.set((a_q2_incorrect | a_q2_correct).ids)],
+            'triggering_answer_ids': [Command.set((a_q1_partial | a_q1_incorrect).ids)],
+            'constr_mandatory': False,
+        }, {
+            'survey_id': survey.id,
+            'title': 'Are you sure?',
+            'sequence': 5,
+            'question_type': 'simple_choice',
+            'suggested_answer_ids': [Command.set((a_q3_correct | a_q3_incorrect).ids)],
+            'triggering_answer_ids': [Command.set([a_q1_correct.id])],
+            'constr_mandatory': False,
+        }])
+        pages = [
+            {'is_page': True, 'question_type': False, 'sequence': 1, 'title': 'Page 0', 'survey_id': survey.id},
+            {'is_page': True, 'question_type': False, 'sequence': 4, 'title': 'Page 1', 'survey_id': survey.id},
+        ]
+
+        q1_correct_answer = {str(q1.id): [a_q1_correct.id]}
+        cases = [
+            ('page_per_question', [], q1_correct_answer),
+            ('page_per_question', a_q1_correct, q1_correct_answer),
+            ('page_per_question', a_q1_incorrect, q1_correct_answer),
+            ('one_page', [], q1_correct_answer), # skipping gives answers for active questions (q2 and q3 conditional questions are inactive)
+            ('one_page', a_q1_correct, {**q1_correct_answer, str(q3.id): [a_q3_correct.id]}),
+            ('one_page', a_q1_partial, {**q1_correct_answer, str(q2.id): [a_q2_correct.id]}),
+            # page0 contains q1 and q2, page1 contains q3
+            ('page_per_section', [], q1_correct_answer),
+            ('page_per_section', a_q1_correct, q1_correct_answer), # no correct answers for q3 because q3 is not on the same page as q1
+            ('page_per_section', a_q1_partial, {**q1_correct_answer, str(q2.id): [a_q2_correct.id]}),
+        ]
+
+        for layout, answer_q1, expected_correct_answers in cases:
+            with self.subTest(layout=layout):
```

Comment:
You could `enumerate` cases above so you get an index you can use as parameter to subtest. Otherwise, you're going to have 3 subtests with the same label.